### ISSUE TITLE
test: Sepearete stop and clean in TestFramework

### ISSUE THREAD

**Issue by rajarshimaitra** [utxo-teleport/teleport-transactions#86]
Currently we do the backend bitcoind stopping and cleaning up the datadir at single method.

https://github.com/utxo-teleport/teleport-transactions/blob/9ec7d4547be2a7afe42032fa13bddee11c6b89ca/src/test_framework.rs#L184-L195

the cleaning up should be taken to another independent method `clean()`.

Each test should then end with 

```rust
// Stop bitcoind
test_framework.stop();

// Clean datadir, comment it out if you need data persistence.
test_framework.clean();
```

**Comment by rajarshimaitra** [comment]
Upon further thought, a better approach is

- use the system's `/tmp` dirs for the test files.
- remove the cleanup code.
- No need for a separate `clean()` method.

**Comment by Shourya742** [comment]
Resolved in #85
